src/app/services/constraint.service.ts

```diff
@@ -407,10 +417,33 @@ export class ConstraintService {
     }
   }
 
+  public isVariableNode(treeNodeType: string): boolean {
+    return (treeNodeType === 'NUMERIC' ||
+      treeNodeType === 'CATEGORICAL' ||
+      treeNodeType === 'CATEGORICAL_OPTION' ||
+      treeNodeType === 'DATE' ||
+      treeNodeType === 'HIGH_DIMENSIONAL' ||
+      treeNodeType === 'TEXT')
+  }
+
   /**
    * Generate the constraint based on the variables selected in the Variables panel
    */
   public variableConstraint(): Constraint {
+    if (this.variablesViewMode === VariablesViewMode.TREE_VIEW) {
```

Comment:
Is it not possible to store the selected variables in one place? Then also your selection is maintained when switching view mode.